src/vcBPA.cpp

```diff
@@ -923,87 +920,94 @@ vdkError vcBPA_ConvertReadPoints(vdkConvertCustomItem *pConvertInput, vdkPointBu
   // Reset point count to avoid infinite loop
   pBuffer->pointCount = 0;
 
-  vcBPAConvertItem *pData = (vcBPAConvertItem *)pConvertInput->pData;
-  bool getNextGrid = true;
+  while (pBuffer->pointCount == 0)
+  {
+    vcBPAConvertItem *pData = (vcBPAConvertItem *)pConvertInput->pData;
+    bool getNextGrid = true;
 
-  uint32_t displacementOffset = 0;
-  vdkError error = vE_Failure;
+    uint32_t displacementOffset = 0;
+    vdkError error = vE_Failure;
 
-  static int gridCount = 0;
-  if (pData->activeItem.pointIndex == 0)
-    ++gridCount;
+    static int gridCount = 0;
+    if (pData->activeItem.pointIndex == 0)
+      ++gridCount;
 
-  if (pData->activeItem.pointIndex == 0 && pData->activeItem.pGrid == nullptr)
-    goto epilogue;
+    if (pData->activeItem.pointIndex == 0 && pData->activeItem.pGrid == nullptr)
+      goto epilogue;
 
-  error = vdkAttributeSet_GetOffsetOfNamedAttribute(&pBuffer->attributes, "udDisplacement", &displacementOffset);
-  if (error != vE_Success)
-    return error;
+    error = vdkAttributeSet_GetOffsetOfNamedAttribute(&pBuffer->attributes, "udDisplacement", &displacementOffset);
+    if (error != vE_Success)
+      return error;
 
-  for (size_t i = pData->activeItem.pointIndex; i < pData->activeItem.pGrid->vertices.length; ++i)
-  {
-    if (pBuffer->pointCount == pBuffer->pointsAllocated)
+    for (size_t i = pData->activeItem.pointIndex; i < pData->activeItem.pGrid->vertices.length; ++i)
     {
-      getNextGrid = false;
-      pData->activeItem.pointIndex = (uint32_t)i;
-      break;
-    }
+      if (pBuffer->pointCount == pBuffer->pointsAllocated)
+      {
+        getNextGrid = false;
+        pData->activeItem.pointIndex = (uint32_t)i;
+        break;
+      }
 
-    udDouble3 position = pData->activeItem.pGrid->vertices[i].position;
-    size_t triangle = vcBPA_FindClosestTriangle(&pData->activeItem.oldGrid, position);
+      udDouble3 position = pData->activeItem.pGrid->vertices[i].position;
 
-    double distance = FLT_MAX;
-    if (triangle < pData->activeItem.oldGrid.triangles.length)
-      distance = udAbs(vcBPA_DistanceToTriangle(&pData->activeItem.oldGrid, triangle, position));
+      if (!udPointInAABB(position, pData->activeItem.pGrid->minPos, pData->activeItem.pGrid->maxPos))
+          continue;
```

Comment:
What bug does this fix?

Reply:
multiple inclusion of points that were outside the AABB for the sake of the algorithm to complete boundary triangles

Reply:
`pGrid` shouldn't contain any points outside the AABB - we don't mesh `pGrid`, we mesh `pOldGrid`. If `pGrid` contains any points outside the AABB, that should be fixed instead.